service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/cmo/ReviewCMOController.java

```diff
@@ -107,6 +116,12 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
                 sealedCMOs.add(cmoToSeal);
 
                 data.put("sealedCMOs", sealedCMOs);
+
+                if (featureToggleService.isNewCaseStateModelEnabled()
+                    && caseData.getReviewCMODecision().hasReviewOutcomeOf(SEND_TO_ALL_PARTIES)
+                    && isNextHearingOfType(caseData, cmo.getId(), ISSUE_RESOLUTION)) {
+                    data.put("state", State.ISSUE_RESOLUTION.getValue());
+                }
```

Comment:
Ok I see why now different tests on the controller - that's doing too much work in the controller. 
Controller should read and convert the data not make business decisions.